doc/dev.md

```diff
@@ -0,0 +1,30 @@
+# Developer documentation
+
+## GHC versions
+
+We support the three most recent versions of GHC.
+We try to support new versions as soon as they are supported by the libraries that we depend on.
+
+### Adding a new version
+
+The following checklist enumerates the steps needed to support a new version of GHC.
+When performing such an upgrade, it can be helpful to copy/paste this list into the MR description and check off what has been done, so as to not forget anything.
+
+- [ ] Allow the [new version of `base`][base] in the Cabal `build-depends`
+- [ ] Run `cabal {build,test,haddock}`, bumping dependency bounds as needed
+- [ ] Fix any new warnings from [`-Wdefault`][wdefault]
+- [ ] Run [`cabal freeze`][freeze] and rename `cabal.freeze` to `cabal.GHC-X.Y.Z.config`
+- [ ] Add the new GHC version to the matrix in the GitHub Actions workflows
+- [ ] Add the new version to the code that sets `GHC_NIXPKGS` in the CI config
+- [ ] Use the new GHC version in the Dockerfiles in `.github/`
+- [ ] Follow the below steps to remove the oldest GHC version
+
+[base]: https://www.snoyman.com/base/
+[freeze]: https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-freeze
+[wdefault]: https://downloads.haskell.org/ghc/latest/docs/users_guide/using-warnings.html#ghc-flag-Wdefault
+
+### Removing an old version
+
```

Comment:
Add para:  This enumerates the steps for removing the support for an older version of GHC that we decide will no longer be supported.  We will continue to support older versions until the point where the maintenance becomes burdensome (e.g. significant `#ifdef` in the code, `if` clauses in the cabal file, etc.), although we commit to supporting the latest three versions regardless of the burden.

Reply:
See [this discussion](https://github.com/GaloisInc/crucible/pull/968/files#diff-9ad82307f9850d8d79f949bbd805f3f53ca211fd33b2d808bb9aed7dcde52aecL47) for deciding what we want the policy to be.